pkg/util/deeptest/statictesthelper.go

```diff
@@ -157,6 +157,9 @@ func (h *staticTestHelper) assertDeepClonedEqual(t require.TestingT, valA, valB
 			h.assertDeepClonedEqual(t, valA.Elem(), valB.Elem(), path)
 		}
 	case reflect.Slice:
+		if valA.IsNil() && valB.IsNil() {
```

Comment:
For slice, map, and function, they can just be recognized as deep cloned when comparing nil